### DISCUSSION TITLE
Configurable dashboards

### DISCUSSION THREAD

**Discussion by marcklingen** [langfuse/langfuse#1011] [Ideas]
Langfuse includes a lot of interesting data (dimensions and metrics). Currently Langfuse includes a number of prebuilt charts on the main dashboard.

We consider building configurable dashboards to plot (timeseries) or pivot (table) any of the data ingested into Langfuse. Creating this issue to understand overall excitement for this and collect use cases. Please share feedback/ideas :)

## Dimensions

- Model
- Version/Release (Experimentation)
- Dataset Run
- User
- Use case (name of trace/observation)

_not exhaustive_

## Metrics

- Volume
- Cost
- Usage (eg tokens)
- Latency: overall, time-to-first-token
- Streaming speed: tokens/second
- Error rates
- Scores: user feedback, model based evals

_not exhaustive_

## EDIT 2025-04-09

A beta preview for custom dashboards is available. You can find the relevant documentation here: https://langfuse.com/docs/analytics/custom-dashboards. Let us know what you think 🙌

**Comment by ulan-yisaev** [comment]
I was about to request a similar feature. It would be fantastic to see this implemented!

**Reply by JanEisen** [reply]
Absolutely agree!

**Reply by ulan-yisaev** [reply]
Hi Marc,

I'd like to share some ideas regarding use cases for this 'Configurable dashboards' feature:

1. User Satisfaction Metrics: A widget to track the percentage of users satisfied or unsatisfied with the LLM app interactions. The ability to define and compare different time periods (e.g., day-to-day, month-to-month) would be crucial.
2. Peak Usage Times Analysis: A feature to identify the times when the app is most frequently used, displayed as hourly granulated bar charts. Including an option to group certain hours (like 21:00 to 06:00) would be beneficial for understanding user behavior patterns.
3. Query Volume Tracking: The ability to monitor the number of inquiries received within selectable time frames. This would help in analyzing trends and peak periods of app interactions.
4. Average Messages per Session: A widget to calculate and display the average number of messages exchanged in each session. This metric would provide insights into the depth of interactions and user engagement.
5. Topic Trend Visualization: Implementing a feature to identify and visually represent frequently discussed topics, potentially through a word cloud. Including a filter for selecting specific time periods would enable targeted analysis of user interests.

IMHO, these features would greatly enhance anyone's ability to analyze and improve their LLM app's performance and user experience. Looking forward to seeing how these ideas can be integrated into the Langfuse dashboards 😊

Thanks for considering these suggestions!

**Comment by norton120** [comment]
I am looking at trend reporting solutions for dataset scores - we run an experiment suite against a dataset for each release, then need to aggregate and present that data (current plan is s3+athena+metabase for a quick/dirty evaluation trends report). It would be very cool to have this baked into Langfuse already, and makes sense as the datasets and scores already reside there

**Comment by yoni-joon** [comment]
I think that's a super important initiative!
in addition to the dimensions described above, i think it'd be super useful to also allow custom dimensions, for users to define themselves. This can allow smart breakdowns over data points.

Best example I can think of is RAG scenario, where the model responds with an `I don't know`; it'd be super helpful to breakdown according to whether context was fetched or not(or even whether GOOD context was fetched), in order to identify the problem

**Comment by Thirunayan22** [comment]
This would be a really useful feature!

**Comment by DeniDoman** [comment]
During the LangFuse features research I face the lack of Evaluation dashboard. Now it exists only as a part of a combined (main) dashboard. I'd say this is less convenient than a dedicated eval observation space with enhanced eval analysis tooling. 

Configurable dashboards can partially solve the observability issue, especially if it will be possible to see the custom eval metrics reported via API. 

One small requirement from me: make these dashboard shareable (like traces). 

P.S. Thank you for the product and for open sourcing it!

**Reply by marcklingen** [reply]
Thanks for sharing! very much looking forward to the new dashboarding capabilities. Publishing is a great idea!

**Comment by Steffen911** [comment]
Hey all,
We added a Beta preview of customizable dashboards on Langfuse Cloud. You can find the documentation here: https://langfuse.com/docs/analytics/custom-dashboards. We're looking forward to your feedback 🙌

**Reply by emilyrohrbough** [reply]
It would be great to be able to filter & fine-tune by metadata like you can when filtering traces!

**Reply by wudideyyf** [reply]
Will the "Self-Hosted" version be considered for support in the future?

**Reply by glc-pdossantos** [reply]
Thanks for this. It would be nice to be able to persist filters directly on the dashboard : 
we use one tag per project and we would like to reuse same widgets on different dashboards and put the filter on the dashboard instead of the widget :)

**Reply by Steffen911** [reply]
@wudideyyf Yes, we intend to release a self-hosted option within the coming days.

**Comment by rekram1-node** [comment]
Would love the ability to filter based on scores (filter based on score name and stuff like that), and then building dashboards for specific datasets would be nice too.

**Reply by Steffen911** [reply]
@rekram1-node Thank you for the feedback! We plan a couple of improvements in the coming weeks to move it out of beta and we'll include this there. I'll ping you here once that is available.
Anything aside from name (and likely value) that come to mind?

**Reply by rekram1-node** [reply]
@Steffen911 Regarding scores no, but I have lots of wants for dataset run dashboards, I think for now we are planning on exporting all the runs, converting to csv, and charting using some third party tools and doing that is a bit of a pain, would love an easier way to setup dashboards for specific datasets in an easier fashion.

**Reply by Steffen911** [reply]
@rekram1-node Could you share which metrics would be most interesting to you? As of today, our dataset runs are not within ClickHouse which will make dashboards a bit challenging, but we're investigating changes there. If you could share your needs that would help us evaluate and prioritize this.

In case you already have a couple of charts you could show that would be amazing.

**Reply by rekram1-node** [reply]
I don't have the charts I would like to have setup anywhere currently. But something we do often is run our dataset with a variety of models and then we can obviously compare the average score between the runs with different models, but it would be great to see which questions are failing the most across runs, which questions fail only for a specific model seemingly, stuff like that. 

It would be good to see observation level metrics between dataset runs, etc. 

I will add more as I think of it but right now I feel like the evaluation dashboards leave a lot to be desired, and even improving the little existing dashboards I can view for a dataset would be good, but right now any change I make to those isn't persisted. I haven't checked if there is an open bug report for that or not but even that would be an improvement

**Reply by rekram1-node** [reply]
It would also be great to have an easier way to toggle from Avg to P90, etc, from a dashboard rather than for each widget individually, or at least without having to edit the widget and instead maybe just update a toggle on the widget in the dashboard view

**Reply by panagiotisrit** [reply]
Adding a +1 here to state that it would be really nice to have a functionality where dataset-level aggregates metrics can be visualized in the dashboards.

For example:  if I run an experiment with an LLM over a classification task with a specific prompt version, I would like to have an e.g bar chart, where the f1 scores are laid out across prompt versions ran against a dataset experiment. :)

**Comment by marcklingen** [comment]
This is GA now: https://langfuse.com/changelog/2025-05-21-custom-dashboards

s/o to @Steffen911 for making this happen!
